tests/test_api_hosts_delete.py

```diff
@@ -153,6 +153,51 @@ def test_delete_hosts_using_filter(
     assert len(new_hosts) == remaining_hosts.count()
 
 
+def test_delete_all_hosts(
+    event_producer_mock, db_create_multiple_hosts, db_get_hosts, api_delete_all_hosts, patch_xjoin_post
+):
+    created_hosts = db_create_multiple_hosts(how_many=len(XJOIN_HOSTS_RESPONSE_FOR_FILTERING["hosts"]["data"]))
+    host_ids = [str(host.id) for host in created_hosts]
+
+    # set the new host ids in the xjoin search reference.
+    resp = deepcopy(XJOIN_HOSTS_RESPONSE_FOR_FILTERING)
+    for ind, id in enumerate(host_ids):
+        resp["hosts"]["data"][ind]["id"] = id
+    response = {"data": resp}
+
+    # Make the new hosts available in xjoin-search to make them available
+    # for querying for deletion using filters
+    patch_xjoin_post(response, status=200)
+
+    new_hosts = db_create_multiple_hosts()
+    new_ids = [str(host.id) for host in new_hosts]
+
+    # delete hosts using the IDs supposedly returned by the query_filter
+    response_status, response_data = api_delete_all_hosts({"delete_all": True, "confirm_delete_all": True})
+
+    assert '"type": "delete"' in event_producer_mock.event
```

Comment:
It would also be good to verify that the event producer was used the correct number of times, since the above only tells you that at least one "delete" event was written. As we expect all of the hosts you created to be deleted, it would look something like this:
```suggestion
    assert '"type": "delete"' in event_producer_mock.event
    assert event_producer_mock.write_event.call_count == len(created_hosts)
```

Reply:
the `event_producer_mock` does not have `call_count` for the write event but the `delete count` is available in response data; so using `assert response_data.get('hosts_deleted') == len(created_hosts)`

Reply:
Ah, you're right... you'd need to do `mocker.patch.object(event_producer_mock, "write_event")` before calling the endpoint, but then you wouldn't be able to check the value of `event`. In that case, I'm resolving this, and we can deal with `write_event` counts in the Kafka PR